### ISSUE TITLE
diff results in InvalidOperationException

### ISSUE THREAD

**Issue by msymons** [CycloneDX/cyclonedx-cli#72]
Using 0.10.2,  I have two BOMs where using the diff command with `--component-versions` option works fine when comparing B to A but throws an error when comparing A to B  (which happens to be the diff I am seeking!).  The BOMs are from the same project and are generated by CycloneDX Maven plugin 2.2.0

```
$ ./cyclonedx-win-x86.exe diff  --component-versions TES-1-bom.xml TES-2-bom.xml
Unhandled exception: System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at CycloneDX.CLI.Program.Diff(String fromFile, String toFile, InputFormat fromFormat, InputFormat toFormat, StandardOutputFormat outputFormat, Boolean componentVersions)
   at System.CommandLine.Invocation.CommandHandler.GetResultCodeAsync(Object value, InvocationContext context)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseErrorReporting>b__21_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__22_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseDirective>b__20_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseDebugDirective>b__11_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

BOMs that give the error sent via Slack.

**Comment by coderpatros** [comment]
I have a fix for this in PR #73. Although I'm not really happy with the output.

The offending component will be listed twice separately instead of being grouped together. i.e.

```
- component@1.0
+ component@1.2

... more component changes ...

- component@1.1
```

Grouping the output will require some rework that I'll do along with #71

**Comment by coderpatros** [comment]
Fix in v0.10.3
